src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,38 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Task that with a deadline.
+ */
+public class Deadline extends Task {
+    protected final String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Creates a new Deadline task.
+     * @param description Description of the task.
+     * @return Deadline task.
+     * @throws DateTimeParseException If the date is in the wrong format.
+     */
+    public static Deadline createDeadline(String description) throws DateTimeParseException {
+        String[] split = description.split("/by ");
+        LocalDate d1 = LocalDate.parse(split[1]);
+        return new Deadline(split[0], d1.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
+    }
```

Comment:
Would it be better to make the Deadline constructor private if there is going to be a factory method to create Deadline tasks?